url1d5e9d990f_Telsis_jOCP_tgz-pJ8-com_telsis_jocp_sampleApp_BasicOCPAppJ8/src/com/telsis/jocp/messages/TelsisHandlerWithParty.java

```
@@ -0,0 +1,196 @@
+ * limits:
+ * <ul>
+ * <li>The fastSSP is limited to an overall message size of 255 bytes.</li>
+ * <li>The fastIP only supports up to 64 bytes of custom data.</li>
+ * <li>The fastSSP (OCP Caller) only supports up to 128 bytes of custom data.</li>
+ * <li>The fastTC (INAP Interface) supports 233 bytes of custom data.</li>
+ * <li>The fastSCP supports up to 448 bytes of custom data.</li>
+ * <li>The jOCP library supports up to 448 bytes of custom data</li>
+ * </ul>
+ * </td>
+ * </tr>
+ * </table>
+ *
+ * @see TelsisHandlerResult
+ * @author Telsis
+ */
+public class TelsisHandlerWithParty extends TelsisHandler {
+    /** The message type. */
+    public static final LegacyOCPMessageTypes TYPE =
+            LegacyOCPMessageTypes.TELSIS_HANDLER_WITH_PARTY;
+
+    /** The minimum expected length of this message. */
+    private static final int            EXPECTED_LENGTH_MIN = 8;
+    /** The maximum expected length of this message. */
+    private static final int            EXPECTED_LENGTH_MAX = 456;
+
+    /** The maximum length of the payload data. */
+    private static final int            MAX_DATA_LENGTH     = 448;
+
+    /** The party the handler is destined for. */
+    private short                       fwdLegID;
+
+    /**
+     * Decode the buffer into a Telsis Handler With Party message.
+     *
+     * @param buffer
+     *            the message to decode
+     * @throws OCPException
+     *             if the buffer could not be decoded
+     */
+    public TelsisHandlerWithParty(final ByteBuffer buffer) throws OCPException {
+        super(buffer, EXPECTED_LENGTH_MIN, EXPECTED_LENGTH_MAX, false); // Need custom decoding
+
+        int dataLength;
+
+        fwdLegID = buffer.getShort();
+        buffer.get(); // for word alignment
+        setZipNumber(buffer.get());
+        short number = buffer.getShort();
+        try {
+            setHandlerNumber(TelsisHandlerNumber.fromShort(number));
+        } catch (Exception e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)